libs/thread/include/hpx/thread/thread.hpp

```diff
@@ -23,20 +23,19 @@
 #include <exception>
 #include <iosfwd>
 #include <mutex>
-#include <utility>
 #include <type_traits>
+#include <utility>
 
 #include <hpx/config/warnings_prefix.hpp>
 
 ///////////////////////////////////////////////////////////////////////////////
-namespace hpx
-{
+namespace hpx {
     ///////////////////////////////////////////////////////////////////////////
     using thread_termination_handler_type =
         std::function<void(std::exception_ptr const& e)>;
     void set_thread_termination_handler(thread_termination_handler_type f);
 
-    class thread
+    class HPX_EXPORT thread
```

Comment:
What's the rationale of `HPX_EXPORT`ing this class?

Reply:
As far as I can remember it has to be exported (because of the member functions defined in the cpp file). I can have a go at removing it though.

Reply:
Yep, it's required.